kb/2022-04-06/nic_naming_scheme.md

```diff
@@ -0,0 +1,78 @@
+---
+title: NIC Naming Scheme
+descripion: NIC Naming Scheme Change
+slug: nic-naming-scheme
+authors:
+  - name: Date Huang
+    title: Software Engineer
+    url: https://github.com/tjjh89017
+    image_url: https://github.com/tjjh89017.png
+tags: [network]
+hide_table_of_contents: false
+---
+
+# NIC Naming Scheme
+
+## NIC Naming Scheme is changed after upgrade v1.0.1
+
+`systemd` in OpenSUSE Leap 15.3 which is the base OS of Harvester is upgraded to `246.16-150300.7.39.1`. In this version, `systemd` will enable additional nameing scheme `sle15-sp3` which is `v238` with `bridge_no_slot`. When there is a PCI bridge assocated with NIC, `systemd` will never generate `ID_NET_NAME_SLOT` and naming policy in `/usr/lib/systemd/network/99-default.link` will fallback to `ID_NET_NAME_PATH`. According to this change, NIC names might be changed in your Harvester nodes during the upgrade process from `v1.0.0` to `v1.0.1-rc1` or above and it will cause network issues which are assocated with NIC names.
+
+## Effect Settings and Workaround
+
+### Startup Network Configutation
+
+NIC name changes will need to migrate the name in `/oem/99_custom.yaml`. You could use [migration script](https://github.com/harvester/upgrade-helpers/blob/main/hack/udev_v238_sle15-sp3.py) to change the NIC names which is assocated with a PCI bridge.
+
+:::tip
+You could find an identical machine to test naming changes before applying the configuration to production machines
+:::
+
+You could simply execute the script with root account in `v1.0.0` via
+```bash
+# python3 udev_v238_sle15-sp3.py
+```
+
+It will output the patched configuration to screen and you could compare to origin one to ensure there is no exception. (e.g. We could use `vimdiff` to check the configuration)
```

Comment:
```suggestion
It will output the patched configuration to the screen and you could compare it to the original one to ensure there is no exception. (e.g. We could use `vimdiff` to check the configuration)
```